From patchwork Wed Apr 24 04:50:59 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10914069
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id C5DE71575
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 24 Apr 2019 04:51:21 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A617128A0B
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 24 Apr 2019 04:51:21 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 808A128A10; Wed, 24 Apr 2019 04:51:21 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 053C728A0B
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 24 Apr 2019 04:51:20 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726151AbfDXEvT (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 24 Apr 2019 00:51:19 -0400
Received: from mail-pf1-f195.google.com ([209.85.210.195]:33639 "EHLO
        mail-pf1-f195.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725958AbfDXEvT (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 24 Apr 2019 00:51:19 -0400
Received: by mail-pf1-f195.google.com with SMTP id h5so8635600pfo.0
        for <kvm@vger.kernel.org>; Tue, 23 Apr 2019 21:51:18 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=7+MkaZo2VvG4C0DTsxu0vgHCNJWcX2jvStAJTw74hsk=;
        b=mW20Yn8SW4xsHpPtEBVNoPexaAJzUxN8oYdob4czaJRlnZQrWEqnJ4sR8mf/YhQWY1
         oi/UE7jV+UvHOdHvwuUB0zgfmfyacJ+KcmoWZiK7PzkBwZ9Sb+T1tcxHuZ/lb2OiZRPm
         BcgLn8bXcZEznRCdxJQejgpuY16IHpveZZUydBf3CcPjcej9MDDIJotMkytqT0xXAWeV
         Vvis+9+c6qFiiL4oWTBVuWuSOi+QGPtgMQU9cLPmt53kZcE9BPBCXXhCVVkGSCiGmqc3
         bgSAmzpBIE0ej/BXYxlCvv/xQDDUxQTs/hKHGmtOe1KQo80MDSIc6dw0VfCze9LMN1IS
         Hlrw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=7+MkaZo2VvG4C0DTsxu0vgHCNJWcX2jvStAJTw74hsk=;
        b=HRIuXDzdgQqCm+n8swb5vXJO/B8jmNHJi84J954nxkolIFeb9cKV+qmKWyTfEVT2c3
         rNQXOvDTgW6Udm3WGr5+2PHfzWc7o05g399DGpv9qioCA3dWHQFUbnqQf1GNRZA+sTmC
         Ue/9yfCOuTTZ1MM081ffor3hdZdOlwvXvK1CSPzD7U4z4UIFm3u6u2QGwDBmS/uR3baq
         4lOEzDa8nK5E0JWTaDEJuOf+7DnMGyWfRiRpEjtwmk0fBraliSvwVOvSa7JeIFu5Qb7V
         6Dg6VinY1uaxsPuXyPcP1Jk9d1NDkHy1mdCdPlC1j2pb8HC7KaZokIwS7AygGK5/QrGn
         7JNw==
X-Gm-Message-State: APjAAAVbL2rRNqy/NQ2xER5zkVqAQHp56GkN7FVmMCqZL6CEdowgOw8T
        JRCN7hM+n3nDe4a9iCPSPVo=
X-Google-Smtp-Source: 
 APXvYqxZZx+vSidVJojNj1pWpTVXj4uZ2Z0Bj/mpgx7CQVu8h261R/KRQC7bVG0+uViYEZj3MjE3TQ==
X-Received: by 2002:a63:4f41:: with SMTP id p1mr28777482pgl.63.1556081478193;
        Tue, 23 Apr 2019 21:51:18 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 j12sm21487103pgg.79.2019.04.23.21.51.16
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Tue, 23 Apr 2019 21:51:17 -0700 (PDT)
From: nadav.amit@gmail.com
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Nadav Amit <nadav.amit@gmail.com>
Subject: [kvm-unit-tests PATCH] x86: Remove test_multiple_nmi()
Date: Tue, 23 Apr 2019 21:50:59 -0700
Message-Id: <20190424045059.11627-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Nadav Amit <nadav.amit@gmail.com>

According to the discussion in [1], expecting nested NMIs never to be
collapsed is wrong.

[1] https://marc.info/?l=kvm&m=145876994031502&w=2

Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
---
 x86/apic.c | 70 ------------------------------------------------------
 1 file changed, 70 deletions(-)

diff --git a/x86/apic.c b/x86/apic.c
index 51744cf..d1ed5ea 100644
--- a/x86/apic.c
+++ b/x86/apic.c
@@ -334,75 +334,6 @@ static volatile int nmi_received;
 static volatile int cpu0_nmi_ctr1, cpu1_nmi_ctr1;
 static volatile int cpu0_nmi_ctr2, cpu1_nmi_ctr2;
 
-static void multiple_nmi_handler(isr_regs_t *regs)
-{
-    ++nmi_received;
-}
-
-static void kick_me_nmi(void *blah)
-{
-    while (!nmi_done) {
-	++cpu1_nmi_ctr1;
-	while (cpu1_nmi_ctr1 != cpu0_nmi_ctr1 && !nmi_done) {
-	    pause();
-	}
-	if (nmi_done) {
-	    return;
-	}
-	apic_icr_write(APIC_DEST_PHYSICAL | APIC_DM_NMI | APIC_INT_ASSERT, 0);
-	/* make sure the NMI has arrived by sending an IPI after it */
-	apic_icr_write(APIC_DEST_PHYSICAL | APIC_DM_FIXED | APIC_INT_ASSERT
-		       | 0x44, 0);
-	++cpu1_nmi_ctr2;
-	while (cpu1_nmi_ctr2 != cpu0_nmi_ctr2 && !nmi_done) {
-	    pause();
-	}
-    }
-}
-
-static void flush_nmi(isr_regs_t *regs)
-{
-    nmi_flushed = true;
-    apic_write(APIC_EOI, 0);
-}
-
-static void test_multiple_nmi(void)
-{
-    int i;
-    bool ok = true;
-
-    if (cpu_count() < 2) {
-	return;
-    }
-
-    sti();
-    handle_irq(2, multiple_nmi_handler);
-    handle_irq(0x44, flush_nmi);
-    on_cpu_async(1, kick_me_nmi, 0);
-    for (i = 0; i < 1000000; ++i) {
-	nmi_flushed = false;
-	nmi_received = 0;
-	++cpu0_nmi_ctr1;
-	while (cpu1_nmi_ctr1 != cpu0_nmi_ctr1) {
-	    pause();
-	}
-	apic_icr_write(APIC_DEST_PHYSICAL | APIC_DM_NMI | APIC_INT_ASSERT, 0);
-	while (!nmi_flushed) {
-	    pause();
-	}
-	if (nmi_received != 2) {
-	    ok = false;
-	    break;
-	}
-	++cpu0_nmi_ctr2;
-	while (cpu1_nmi_ctr2 != cpu0_nmi_ctr2) {
-	    pause();
-	}
-    }
-    nmi_done = true;
-    report("multiple nmi", ok);
-}
-
 static volatile int lvtt_counter = 0;
 
 static void lvtt_handler(isr_regs_t *regs)
@@ -614,7 +545,6 @@ int main(void)
     test_pv_ipi();
 
     test_sti_nmi();
-    test_multiple_nmi();
 
     test_apic_timer_one_shot();
     test_apic_change_mode();
